src/main/java/com/hivemq/client/mqtt/lifecycle/MqttClientAutoReconnectBuilder.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 dc-square and the HiveMQ MQTT Client Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.hivemq.client.mqtt.lifecycle;
+
+import com.hivemq.client.annotations.DoNotImplement;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Builder for a {@link MqttClientAutoReconnect}.
+ *
+ * @author Silvio Giebl
+ * @since 1.1
+ */
+@DoNotImplement
+public interface MqttClientAutoReconnectBuilder
+        extends MqttClientAutoReconnectBuilderBase<MqttClientAutoReconnectBuilder> {
```

Comment:
Wouldnt an explicit option without exponential backoff be good? Or at least a comment (min = max)

Reply:
Good idea.
As this could be added later, we could create a separate ticket for that.

Reply:
I feel it might be dangerous to have the option for non-exponential backoff. Exponential backoff protects people from DDoS'ing their infrastructure. I've seen this multiple times and if someone *really* knows what she's doing, she could just implement their own reconnect strategy

Reply:
that is true